apps/prairielearn/assets/scripts/question.ts

```diff
@@ -120,7 +120,10 @@ function fetchResults(socket, submissionId) {
         answerContainer.closest('.grading-block').classList.remove('d-none');
       }
       if (msg.submissionPanel) {
-        document.getElementById('submission-' + submissionId).outerHTML = msg.submissionPanel;
+        // Using jQuery here because msg.submissionPanel may contain scripts
+        // that must be executed. Typical vanilla JS alternatives don't support
+        // this kind of script.
+        $('#submission-' + submissionId).replaceWith(msg.submissionPanel);
```

Comment:
For future reference, here's the part of jQuery that seems to handle script execution in `replaceWith`: https://github.com/jquery/jquery/blob/8be4c0e4f89d6c8f780e5937a0534921d8c7815e/src/manipulation/domManip.js